crates/ruff_linter/src/rules/pycodestyle/rules/logical_lines/extraneous_whitespace.rs

```diff
@@ -134,12 +134,36 @@ pub(crate) fn extraneous_whitespace(
     autofix_before_punctuation: bool,
 ) {
     let mut prev_token = None;
+    let mut fstrings = 0u32;
 
     for token in line.tokens() {
         let kind = token.kind();
+        match kind {
+            TokenKind::FStringStart => {
+                fstrings += 1;
+                continue;
+            }
+            TokenKind::FStringEnd => {
+                fstrings = fstrings.saturating_sub(1);
+                continue;
```

Comment:
It looks like we're not setting `prev_token` because we're continuing. Is that ok?

Reply:
Good point. I think it's ok but I'll remove it. It might also make sense to remove the `continue;` as pointed out in the other comment.

Reply:
Is the goal to detect _some_ violations in f-strings, but skip others that could be false positives? Or is it to avoid flagging _anything_ in f-strings (as happens today on `main`, IIUC)?

Reply:
The former but only in cases where it's easier to differentiate between usages.